src/Dictionaries/CacheDictionaryStorage.h

```diff
@@ -72,10 +72,11 @@ class CacheDictionaryStorage final : public ICacheDictionaryStorage
 
     SimpleKeysStorageFetchResult fetchColumnsForKeys(
         const PaddedPODArray<UInt64> & keys,
-        const DictionaryStorageFetchRequest & fetch_request) override
+        const DictionaryStorageFetchRequest & fetch_request,
+        std::shared_ptr<IColumn::Filter> default_mask = nullptr) override
```

Comment:
Why do we need `shared_ptr`? Why not just raw pointer `IColumn::Filter *`?

Reply:
Also, please, write a comment what is `default_mask` and how it should be filled

Reply:
Yes, raw pointer works. I do this on purpose to avoid using raw pointer at all. I remember I reviewed some prs just to remove usage of raw pointers (class members). Here raw pointer is used in function parameter. Do you prefer raw pointer or smart one?

Reply:
In our case I would prefer raw pointer because we don't work with `std::shared_ptr<IColumn::Filter>` in any place in the code